src/libraries/System.Net.Security/src/System/Net/Security/SslAuthenticationOptions.cs

```diff
@@ -106,6 +111,12 @@ internal void UpdateOptions(SslServerAuthenticationOptions sslServerAuthenticati
             if (sslServerAuthenticationOptions.ServerCertificateContext != null)
             {
                 CertificateContext = sslServerAuthenticationOptions.ServerCertificateContext;
+
+                if (CertificateContext.Trust != null && sslServerAuthenticationOptions.ChainPolicy != null &&
+                    sslServerAuthenticationOptions.ChainPolicy.TrustMode != X509ChainTrustMode.CustomRootTrust)
+                {
+                    throw new InvalidOperationException("Unable to use custom trust without custom root"); ;
```

Comment:
I think the intent here is that if you specified a trust value and a policy that you have to be in custom root trust on the policy... but it doesn't verify that they're the same contents.

Unless it's going to verify they're the same set, I'd just ignore it.  `Trust` would still control the trust list in the handshake (if opted in), but if there's also a policy then it'll do what the policy says.  (That also solves the problem of what to do about future potential trust modes, like CustomAnchors (stop at intermediates that are in the collection, don't have to be roots) or SystemOrCustomRootTrust (which isn't very likely))

Reply:
That would be ok. We can define what the behavior is.